File: common/src/test/scala/com/taskadapter/connector/common/FileNameGeneratorTest.scala
package com.taskadapter.connector.common

import java.io.File

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class FileNameGeneratorTest extends FunSpec with Matchers {
  it("second file name is different from first when a file already exists") {
    val folder = new File(System.getProperty("java.io.tmpdir"));

    val filePattern = "file_%d.txt"
    val file = FileNameGenerator.findSafeAvailableFileName(folder, filePattern)
    file.createNewFile()
    file.deleteOnExit()

    val file2 = FileNameGenerator.findSafeAvailableFileName(folder, filePattern)
    file2.getName should not be file.getName
    file2.deleteOnExit()
  }
}

File: common/src/test/java/com/taskadapter/connector/common/FileNameGeneratorTest.java
package com.taskadapter.connector.common;

import org.junit.Test;

import java.io.File;
import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

public class FileNameGeneratorTest {
    @Test
    public void secondFileNameIsDifferentFromFirstWhenFileAlreadyExists() throws IOException {
        var folder = new File(System.getProperty("java.io.tmpdir"));

        var filePattern = "file_%d.txt";
        var file = FileNameGenerator.findSafeAvailableFileName(folder, filePattern);
        file.createNewFile();
        file.deleteOnExit();

        var file2 = FileNameGenerator.findSafeAvailableFileName(folder, filePattern);
        assertThat(file2.getName()).isNotEqualTo(file.getName());
        file2.deleteOnExit();
    }
}
